clients/src/main/java/org/apache/kafka/clients/consumer/RangeAssignor.java

```diff
@@ -40,6 +40,29 @@
  * <li><code>C0: [t0p0, t0p1, t1p0, t1p1]</code></li>
  * <li><code>C1: [t0p2, t1p2]</code></li>
  * </ul>
+ *
+ * Since the introduction of static membership, we could leverage <code>group.instance.id</code> to make the assignment behavior more sticky.
+ * For the above example, after one rolling bounce, group coordinator will attempt to assign new <code>member.id</code> towards consumers,
+ * for example <code>C0</code> -> <code>C3</code> <code>C1</code> -> <code>C2</code>.
+ *
+ * <p>The assignment could be completely shuffled to:
+ * <ul>
+ * <li><code>C3 (was C0): [t0p2, t1p2] (before was [t0p0, t0p1, t1p0, t1p1])</code>
+ * <li><code>C2 (was C1): [t0p0, t0p1, t1p0, t1p1] (before was [t0p2, t1p2])</code>
+ * </ul>
+ *
+ * The assignment change was caused by the change of <code>member.id</code> relative order, and could be
+ * mitigated by the introduction of static membership. Consumers will have individual instance ids
+ * <code>I1</code>, <code>I2</code>. As long as
+ * 1. Number of members remain the same across generation
+ * 2. Static members' identities persist across generation
+ * 3. Subscription pattern doesn't change for any member
+ *
+ * <p>The assignment will always be:
+ * <ul>
+ * <li><code>I0: [t0p0, t0p1, t1p0, t1p1]</code>
+ * <li><code>I1: [t0p2, t1p2]</code>
+ * </ul>
  */
```

Comment:
I see your point. The purpose of the comparison is to let user understand that `group.instance.id` will be ephemeral, which could be easier to memorize when seeing an example, and in the same making the motivation clear.

Reply:
You mean `member.id` is ephemeral, don't you?

As far as I understand, if the `group.instance.id` is set the assignment will be more sticky than without. It all boils down to `Collections.sort(consumersForTopic)` and the comparator of `MemberInfo`. If `group.instance.id` is set the comparator will use it, otherwise it will use `member.id`. 

If this is correct, then maybe my comment above was a bit too strong. Still I think you should get rid of the conditional tense. For example, you should write `Since the introduction of static membership, we can leverage <code>group.instance.id</code> to make the assignment behavior more sticky.`, because we do it if `group.instance.id`is set. It's a fact.

Here is another sentence, I would change:
`The assignment change was caused by the change of <code>member.id</code> relative order, and could be mitigated by the introduction of static membership` -> `The assignment change is caused by the change of <code>member.id</code> relative order, and can be avoided by setting the group.instance.id.`